models/interfaces.ts

```diff
@@ -66,3 +79,61 @@ export interface State {
   actions: object[];
   transitions: object[];
 }
+
+export interface Rollup {
+  continuous: boolean;
+  delay?: number;
+  description?: string;
+  dimensions: [
```

Comment:
Updated to make "dimensions: dateHistogramItem | termsItem | histogramItem[]; "

Reply:
That seems like it's saying dimensions is a date histogram item, a terms item, or an array of histogram items?

Reply:
You are right. I actually do have a RollupDimentionItem interface from another interfaces.ts file. Moved it to this interface.ts file.
So it becomes `dimentions: RollupDimentionItem[]`, while 

`RollupDimensionItem {
  date_histogram?: {
    fixed_interval: string | null;
    calendar_interval: string | null;
    source_field: string;
    target_field: string;
    timezone: string;
  };
  terms?: {
    source_field: string;
    target_field: string;
  };
  histogram?: {
    source_field: string;
    target_field: string;
    interval: number;
  };
}`

Made the 3 types optional since it could be either one of it. Still not familiar with the null or optional definition. Let me know if it seems wrong.

Reply:
it's been a while but I think you can just wrap it in parens like` (dh, t, h)[]`
Or you could have RollupDimensionItem[] where rdi is d | t | h where you define each d, t, and h separately. As that current blob you have has {} being valid which it isn't, it must be one of them.

Reply:
So, change to `RollupDimensionItem = (DateHistogramItem |TermsItem | HistogramItem)` should be fine?

Reply:
It's been a while.. but yes I believe so. You could verify on the TS docs